### DISCUSSION TITLE
Capture keyboard and mouse events

### DISCUSSION THREAD

**Discussion by mouralabank** [CommunityToolkit/Maui#1062] [New Feature Discussions]
I often find myself needing to perform certain actions based on specific keyboard and mouse inputs, and I believe this feature would greatly enhance the user experience.

Specifically, I would like to be able to capture events such as:

Key presses (e.g., space bar, control key, number keys)
Mouse clicks (e.g., left-click, right-click)
Mouse movements (e.g., scrolling, hovering)

Having access to these events would enable me to create custom workflows and automate certain tasks within the application. For example, I could use key presses to quickly navigate between different views or perform certain actions.

**Comment by danielancines** [comment]
Nice points we have here, It'll be great have this flexibility to implement some features on Desktop applications.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889

**Comment by mentat9** [comment]
Seriously? Nobody cares about hooking mouse or keyboard events? My app badly needs it (ported from Xamarin, where it is supported/easy), and I've been waiting for MAUI support since MAUI was a thing. Being able to respond to, say an arrow key isn't exactly an exotic feature. My question for the dotnet/MAUI team is also not getting any response:  https://developercommunity.visualstudio.com/t/Need-to-hook-keyboard. -events-in-Windows/10541263. I think MAUI is great, but it's not ready to replace Xamarin when it doesn't support basic stuff like this. And now Xamarin support is going away... feeling like I'm getting left high and dry.

**Reply by VladislavAntonyuk** [reply]
You can use any Global/Local hook lib.

**Reply by mentat9** [reply]
Yeah, guess I should have been on that road already. Thanks for the push, investigating now.

**Comment by danielancines** [comment]
This is so bad for Desktop Maui App experience!

**Reply by mentat9** [reply]
I ported my code over to Sharphook (https://github.com/TolikPylypchuk/SharpHook). There are some bad aspects (particularly behavior under the debugger, which is absolutely awful), but it's working as an OK replacement so far. It has the advantage of being natively cross platform, so you don't need to implement each platform separately, which was needed in Xamarin.

**Reply by danielancines** [reply]
I used sharphook but after facing a lot of issues, I change it for a home maid solution, but this kind of feature is default for desktop applications.

**Comment by mentat9** [comment]
> I used sharphook but after facing a lot of issues, I change it for a home maid solution, but this kind of feature is default for desktop applications.

Agree, it's a really important missing capability for desktop apps. Hopefully the MAUI team will find time to add support before too long.

**Reply by bijington** [reply]
I see while this discussion is closed there has been some activity on it. Is there something as the community toolkit team we can do to support mouse/keyboard events?

**Reply by danielancines** [reply]
Is there any chance to include mouse key down and up events on dotnet Maui? Mouse cursors support will be great too.
